index.js

```diff
@@ -328,7 +331,18 @@ async function parsePipelineTemplate({ yaml }) {
 
     // flatten pipeline template shared setting into jobs
     pipelineTemplateConfig.jobs = flattenSharedIntoJobs(pipelineTemplateConfig.shared, pipelineTemplateConfig.jobs);
+
+    const mergedJobs = {};
+
+    // Handle mergeSharedSteps annotation
+    Object.keys(pipelineTemplateConfig.jobs).forEach(j => {
+        mergedJobs[j] = handleMergeSharedStepsAnnotation({
+            sharedConfig: pipelineTemplateConfig.shared,
+            jobConfig: pipelineTemplateConfig.jobs[j]
+        });
```

Comment:
third one is not required

Reply:
when will 3rd one ever required then? 3rd one does not have a default value, so it will always become undefined when not supplied during function call